biorxiv/article_distances/biorxiv_article_distances.py

```diff
@@ -0,0 +1,499 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Find published articles missing from bioRxiv
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import plotnine as p9
+from scipy.spatial.distance import cdist
+import tqdm
+
+
+# # Load Embeddings
+
+# ## bioRxiv
+
+# In[2]:
+
+
+biorxiv_journal_df = (
+    pd.read_csv(
+        "../journal_tracker/output/mapped_published_doi.tsv", 
+        sep="\t"
+    )
+    .groupby("doi")
+    .agg({
+        "document":"last",
+        "category":"first",
+        "journal":"first",
+        "doi":"last",
+        "published_doi":"first",  
+        "pmcid":"first", 
+        "pmcoa":"first"
+    })
+    .reset_index(drop=True)
+)
+biorxiv_journal_df.head()
+
+
+# In[3]:
+
+
+biorxiv_embed_df = (
+    pd.read_csv(
+        Path("../word_vector_experiment/output/") /
+        "word2vec_output/" /
+        "biorxiv_all_articles_300.tsv.xz",
+        sep="\t"
+    )   
+)
+biorxiv_embed_df.head()
+
+
+# In[4]:
+
+
+biorxiv_journal_mapped_df = (
+    biorxiv_journal_df[["document", "published_doi", "pmcid", "pmcoa"]]
+    .merge(biorxiv_embed_df, on="document")
+)
+biorxiv_journal_mapped_df.head()
+
+
+# ## Pubmed Central
+
+# In[5]:
+
+
+pmc_articles_df = (
+    pd.read_csv(
+        Path("../../pmc/exploratory_data_analysis/") /
+        "output/pubmed_central_journal_paper_map.tsv.xz", 
+        sep="\t"
+    )
+    .query("article_type=='research-article'")
+)
+pmc_articles_df.head()
+
+
+# In[6]:
+
+
+pmc_embed_df = (
+    pd.read_csv(
+        list(
+            Path("../../pmc/word_vector_experiment/output/")
+            .glob("*300.tsv.xz")
+        )[0],
+        sep="\t"
+    )
+)
+pmc_embed_df.head()
+
+
+# In[7]:
+
+
+pmc_journal_mapped_df = (
+    pmc_articles_df[["journal", "doi", "pmcid"]]
+    .merge(pmc_embed_df, left_on="pmcid", right_on="document")
+    .drop("pmcid", axis=1)
+)
+pmc_journal_mapped_df.head()
+
+
+# # Calculate Distances
+
+# ## biorxiv -> published versions
+
+# In[8]:
+
+
+biorxiv_published = (
+    biorxiv_journal_mapped_df
+    .query("pmcid.notnull()")
+    .query("pmcoa == True")
+    .sort_values("pmcid", ascending=True)
+    .drop_duplicates("pmcid")
+    .set_index("pmcid")
+)
+biorxiv_published.head()
+
+
+# In[9]:
+
+
+PMC_pubished = (
+    pmc_journal_mapped_df
+    .query(f"document in {biorxiv_published.reset_index().pmcid.tolist()}")
+    .sort_values("document", ascending=True)
+    .set_index("document")
+)
+PMC_pubished.head()
+
+
+# In[10]:
+
+
+article_distances = cdist(
```

Comment:
This is the distance between all preprints vs all published?

Reply:
Based on the plot below it looks like this is 1-1 distance between preprint and its associated publication

Reply:
> This is the distance between all preprints vs all published?
> Based on the plot below it looks like this is 1-1 distance between preprint and its associated publication

Correct. This is the distance between preprint embeddings and their published counter part embeddings. I'm forcing it to be a 1-1 mapping in this case.